hazelcast-jet-core/src/main/java/com/hazelcast/jet/Processors.java

```diff
@@ -375,11 +377,13 @@ public static ProcessorSupplier streamFiles(@Nonnull String watchedDirectory) {
      *
      * @param watchedDirectory The directory where we watch files
      * @param charset charset to use to decode the files, or {@code null} to use UTF-8
+     * @param glob the file filter, see {@link
+     *             java.nio.file.FileSystem#getPathMatcher(String) getPathMatcher()}
      */
     public static ProcessorSupplier streamFiles(
-            @Nonnull String watchedDirectory, @Nullable Charset charset
+            @Nonnull String watchedDirectory, @Nullable Charset charset, @Nullable String glob
     ) {
-        return StreamFilesP.supplier(watchedDirectory, charset == null ? null : charset.toString());
+        return StreamFilesP.supplier(watchedDirectory, charset == null ? null : charset.toString(), glob);
```

Comment:
We need a consistent story for the whole API. We either do or don't substitute defaults for `nulls`.  Calling with `StandardCharsets.UTF_8` instead of `null` is not much more difficult and makes the choice clear. I'm especially concerned wit calls like `(path, null, null)`.